src/republication.rs

```diff
@@ -49,11 +54,40 @@ impl<'a> RepublicationBuffer<'a> {
         }
     }
 
-    pub fn pop_publish(&mut self, id: u16) -> Result<(), ProtocolError> {
-        let (_, header) = self
+    fn probe_header(&mut self, index: usize) -> Option<(usize, MqttHeader)> {
+        let mut offset = 0;
+
+        if index > 0 {
```

Comment:
The `if` can go and this can become one loop `for _ in ..index + 1 { ... break packet; }` or also move the `Some((offset..))` into the loop as well.

Reply:
You can't break out of for-loops, but I've simplified nonetheless